pkg/cmd/openshift-controller-manager/controller_manager.go

```diff
@@ -56,58 +44,20 @@ func RunOpenShiftControllerManager(config *openshiftcontrolplanev1.OpenShiftCont
 		klog.Infof("Build controller using images from %q", imageTemplate.ExpandOrDie("<component>"))
 	}
 
-	originControllerManager := func(c context.Context) {
-		if err := WaitForHealthyAPIServer(kubeClient.Discovery().RESTClient()); err != nil {
-			klog.Fatal(err)
-		}
-
-		controllerContext, err := origincontrollers.NewControllerContext(c, *config, clientConfig)
-		if err != nil {
-			klog.Fatal(err)
-		}
-		if err := startControllers(controllerContext); err != nil {
-			klog.Fatal(err)
-		}
-		controllerContext.StartInformers(c.Done())
+	if err := WaitForHealthyAPIServer(kubeClient.Discovery().RESTClient()); err != nil {
+		return err
 	}
 
-	eventBroadcaster := record.NewBroadcaster()
-	eventBroadcaster.StartLogging(klog.Infof)
-	eventBroadcaster.StartRecordingToSink(&v1core.EventSinkImpl{Interface: kubeClient.CoreV1().Events("")})
-	eventRecorder := eventBroadcaster.NewRecorder(legacyscheme.Scheme, v1.EventSource{Component: "openshift-controller-manager"})
-	id, err := os.Hostname()
+	originControllerContext, err := origincontrollers.NewControllerContext(ctx, *config, controllerContext.KubeConfig)
 	if err != nil {
 		return err
 	}
-	rl, err := resourcelock.New(
-		resourcelock.LeasesResourceLock,
-		"openshift-controller-manager",
-		"openshift-master-controllers", // this matches what ansible used to set
```

Comment:
This must be now set in the config under leaderElection.name, there is no other way to set this.

Reply:
should be resolved by https://github.com/openshift/cluster-openshift-controller-manager-operator/pull/385